packages/modules/valkey/src/valkey-container.ts

```diff
@@ -0,0 +1,103 @@
+import { AbstractStartedContainer, GenericContainer, StartedTestContainer, Wait } from "testcontainers";
+import path from "path";
+
+const VALKEY_PORT = 6379;
+
+export class ValkeyContainer extends GenericContainer {
+  private readonly importFilePath = "/tmp/import.valkey";
+  private password? = "";
+  private persistenceVolume? = "";
+  private initialImportScriptFile? = "";
+
+  constructor(image = "valkey/valkey:8.0") {
+    super(image);
+    this.withExposedPorts(VALKEY_PORT)
+      .withStartupTimeout(120_000)
+      .withWaitStrategy(Wait.forLogMessage("Ready to accept connections"));
+  }
+
+  public withPassword(password: string): this {
+    this.password = password;
+    return this;
+  }
+
+  public withPersistence(sourcePath: string): this {
+    this.persistenceVolume = sourcePath;
+    return this;
+  }
+
+  public withInitialData(importScriptFile: string): this {
+    this.initialImportScriptFile = importScriptFile;
+    return this;
+  }
+
+  public override async start(): Promise<StartedValkeyContainer> {
+    this.withCommand([
+      "valkey-server",
+      ...(this.password ? [`--requirepass "${this.password}"`] : []),
+      ...(this.persistenceVolume ? ["--save 1 1 ", "--appendonly yes"] : []),
+    ]);
+    if (this.persistenceVolume) {
+      this.withBindMounts([{ mode: "rw", source: this.persistenceVolume, target: "/data" }]);
+    }
+    if (this.initialImportScriptFile) {
+      this.withCopyFilesToContainer([
+        {
+          mode: 666,
+          source: this.initialImportScriptFile,
+          target: this.importFilePath,
+        },
+        {
+          mode: 777,
+          source: path.join(__dirname, "import.sh"),
+          target: "/tmp/import.sh",
+        },
+      ]);
+    }
+    const startedContainer = new StartedValkeyContainer(await super.start(), this.password);
+    if (this.initialImportScriptFile) await this.importInitialData(startedContainer);
```

Comment:
I know that this is copied from the Redis container code, but this can be refactored by moving this line into the `containerStarted` callback

Reply:
Will work for ❤️s.
Sure thing, I'll fix them both up